changes/2024-TBD_key-store-admin/background.md

```diff
@@ -0,0 +1,224 @@
+[//]: # "Copyright Amazon.com Inc. or its affiliates. All Rights Reserved."
+[//]: # "SPDX-License-Identifier: CC-BY-SA-4.0"
+
+# Key Store Administration Client & Configuration
+
+# Definitions
+
+## Conventions used in this document
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+# Background
+
+The parameters needed to fetch Branch Keys for usage (encrypt, decrypt)
+are distinct from the parameters needed to administrate them
+(create, version/rotate, any future operations).
+
+The permissions needed for the persistence medium,
+which is DynamoDB by default,
+and to Key Management Service,
+which currently can only be AWS KMS,
+are also different between usage and administration activities.
+
+As such, it makes sense to create a
+new Branch Key Store Administration Client,
+that cannot facilitate usage,
+only administrative operations.
+
+## Alternatives to a new Client/Local Service?
+
+### Refactor Key Store
+
+Rather than creating a new Client/Local Service,
+Crypto Tools could refactor the existing
+Key Store Client to behave in the manner described below.
+
+This would be a breaking change,
+and is therefore not recommended.
+
+### Static Methods on the MPL
+
+A non-breaking change is to add the behaviors
+described below to the Material Provider's Local Service/Client,
+rather than creating a new Client.
+
+The advantage of such an approach is less software
+maintained and documented by Crypto Tools.
+
+However,
+this approach forces the Material Provider's Client
+to directly reference DynamoDB.  
+In the long run,
+Crypto Tools MAY want allow MPL Consumers to optionally
+depend on the AWS SDK,
+as the original AWS Encryption SDK afforded that.
+Adding a direct reference to DynamoDB
+to the Material Provider's Client
+complicates such an ambition.
+
+# Name of the Client?
+
+It is best that we bike shed on this early.
+
+The proposed name for this client is
+`the Key Store Admin Client`
+though we may often refer to it as
+just the `Key Store Admin`.
+
+# Proposed Construction
+
+The Key Store Admin requires the following arguments:
+
+- Storage Reference
+- Logical Key Store Name
+
+See [Key Store Persistence Changes](changes/2024-6-17_key-store-persistance/background.md) for details
+on Storage Reference.
+
+# Common Parameters and their defaults
+
+## `KMS Identifier`
+
+`KMS Identifier` is a KMS ARN which identifies the KMS Key
+that will be used for the operation.
+This ARN MUST NOT be an Alias.
+This ARN MUST be a valid
+[AWS KMS Key ARN](./aws-kms/aws-kms-key-arn.md#a-valid-aws-kms-arn).
+This ARN MAY be a Multi-Region Key (MRK) or Single Region Key.
+
+## Key Management
```

Comment:
bikeshed: maybe Key Manager?

Reply:
@justplaz If you want to Bike Shed on this name, you need to head over to @seebees PR:
https://github.com/aws/aws-cryptographic-material-providers-library/pull/594

This just inherits that name.